### ISSUE TITLE
Q: What causes scipy/special/cython_special.c to be regenerated?

### ISSUE THREAD

**Issue by pvanmulbregt** [scipy/scipy#8698]
What dependencies are being used to decide when to regenerate `scipy/special/cython_special.c`?

I added a function to scipy/special/cephes/kolmogorov.c, added entries to cephes.h, cephes_names.h, add_newdocs.py, _cephes.pxd,functions.json, _legacy.pxd, tests/test_cython_special.py, built, made some modifications, rebuilt and ran tests.  But tests/test_cython_special.py failed.
[TypeError in  test_cython_api(param)]

I tracked it down to `special/cython_special.c` being stale.  Whenever I changed functions.json or _cephes.pxd or _legacy.pxd, the Cython file cython_special.c wasn't being updated. 
In my situation I could work around it by modifying `special/_generate_pyx.py` to treat those files as dependencies:
```
     pwd = os.path.dirname(__file__)
     src_files = (os.path.abspath(__file__),
-                 os.path.abspath(os.path.join(pwd, 'add_newdocs.py')))
+                 os.path.abspath(os.path.join(pwd, 'add_newdocs.py')),
+                 os.path.abspath(os.path.join(pwd, 'functions.json')),
+                 os.path.abspath(os.path.join(pwd, '_cephes.pxd')),
+                 os.path.abspath(os.path.join(pwd, '_legacy.pxd'))
+                 )
     dst_files = ('_ufuncs.pyx',
                  '_ufuncs_defs.h',
                  '_ufuncs_cxx.pyx',
```
but I suspect that it's not the true fix.

What should the dependencies be for  `scipy/special/cython_special.c` and how are they managed?
 
### Scipy/Numpy/Python version information:
```
1.1.0.dev0+03f2872 1.13.1 sys.version_info(major=3, minor=6, micro=3, releaselevel='final', serial=0)
```

**Comment by pv** [comment]
It's enough to add the functions.json to src_files, the pxd/pyx/c files are not inputs.
This was just forgotten in gh-8457

**Comment by pv** [comment]
fix in f1424309d892153209f7e609153aa771fd5b8bbd
